src/middlewared/middlewared/test/integration/assets/roles.py

```diff
@@ -53,16 +52,18 @@ def common_checks(
                 with pytest.raises(Exception) as exc_info:
                     client.call(method, *method_args, **method_kwargs)
 
-                assert isinstance(exc_info.value, ClientException) is False or (
-                    exc_info.value.errno != errno.EACCES and exc_info.value.error != 'Not authorized'
+                assert not (
+                    isinstance(exc_info.value, CallError) and
```

Comment:
Is there a reason we're going from `isinstance or (value.errno and value.error)` to `(isinstance and value.errno and value.error)`?

Reply:
Readability

Reply:
We explicitly want the logic that is present now: any exception besides `CallError` that has `errno == errno.EACCES and errmsg == 'Not authorized'`. The old expression was technically incorrect, in practice it didn't affect the tests.

Reply:
Gotcha! Thank you for the clarification